sycl/pi_win_proxy_loader/pi_win_proxy_loader.cpp

```diff
@@ -165,8 +166,25 @@ __declspec(dllexport) void *getPreloadedPlugin(
 
   MapT &dllMap = getDllMap();
 
-  auto match = dllMap.find(PluginPath); // result might be nullptr (not found),
-                                        // which is perfectly valid.
+  // All entries in the dllMap have the same parent directory.
+  // To avoid case sensivity issues, we don't want to do string comparison but
+  // just make sure that directory of the entires in the map and directory of
+  // the PluginPath are equivalent (point to the same physical location).
+  auto match = dllMap.end();
+  std::error_code ec;
+  if (!dllMap.empty() &&
+      std::filesystem::equivalent((dllMap.begin())->first.parent_path(),
+                                  PluginPath.parent_path(), ec)) {
+    // Now we can search only by filename. Result might be nullptr (not found),
+    // which is perfectly valid.
+    match =
+        std::find_if(dllMap.begin(), dllMap.end(),
+                     [&](const std::pair<std::filesystem::path, void *> &v) {
+                       std::error_code ec;
```

Comment:
is `ec` unused?

Reply:
It is used only to pass it std::filesystem::equivalent - this is the version of std::filesystem::equivalent which doesn't throw exception and returns false if either of directories not equivalent or not found.

Reply:
Indeed, removed, thanks!